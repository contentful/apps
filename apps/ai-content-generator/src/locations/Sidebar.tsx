import { SidebarAppSDK } from '@contentful/app-sdk';
import { Paragraph } from '@contentful/f36-components';
import { /* useCMA, */ useSDK } from '@contentful/react-apps-toolkit';

const Sidebar = () => {
  const sdk = useSDK<SidebarAppSDK>();
  /*
     To use the cma, inject it as follows.
     If it is not needed, you can remove the next line.
  */
  // const cma = useCMA();

  return <Paragraph>Hello Sidebar Component (AppId: {sdk.ids.app})</Paragraph>;
};

export default Sidebar;
