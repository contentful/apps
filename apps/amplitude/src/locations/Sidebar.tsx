import { useEffect, useState } from 'react';
import { Button, Paragraph, Stack, Badge, Card } from '@contentful/f36-components';
import { SidebarAppSDK } from '@contentful/app-sdk';
import { useAutoResizer, useSDK } from '@contentful/react-apps-toolkit';

const Sidebar = () => {
  console.log('📊 Amplitude Sidebar: Initializing analytics sidebar');
  
  const sdk = useSDK<SidebarAppSDK>();
  const [isTracking, setIsTracking] = useState(false);
  const [eventCount, setEventCount] = useState(0);
  const [lastEventTime, setLastEventTime] = useState<string | null>(null);

  useAutoResizer();

  console.log('🔗 Amplitude Sidebar: SDK initialized with entry ID:', sdk.entry.getSys().id);
  console.log('🏷️ Amplitude Sidebar: Entry content type:', sdk.contentType.sys.id);

  useEffect(() => {
    console.log('⚡ Amplitude Sidebar: Setting up entry change listeners...');
    
    const unsubscribe = sdk.entry.onSysChanged(() => {
      console.log('📝 Amplitude Sidebar: Entry system data changed');
      trackEntryUpdate();
    });

    return () => {
      console.log('🧹 Amplitude Sidebar: Cleaning up entry listeners');
      unsubscribe();
    };
  }, [sdk.entry]);

  const trackEntryUpdate = () => {
    console.log('📈 Amplitude Sidebar: Tracking entry update event...');
    
    const eventData = {
      eventType: 'content_entry_updated',
      entryId: sdk.entry.getSys().id,
      contentType: sdk.contentType.sys.id,
      timestamp: new Date().toISOString(),
      userId: sdk.user.sys.id,
    };

    console.log('📊 Amplitude Sidebar: Event data prepared:', eventData);
    
    // Simulate tracking (in real app, this would call Amplitude API)
    setEventCount(prev => {
      const newCount = prev + 1;
      console.log('📊 Amplitude Sidebar: Event count updated to:', newCount);
      return newCount;
    });
    
    setLastEventTime(new Date().toLocaleTimeString());
    console.log('✅ Amplitude Sidebar: Entry update event tracked successfully');
  };

  const handleTrackPageView = async () => {
    console.log('👁️ Amplitude Sidebar: Manual page view tracking initiated...');
    setIsTracking(true);
    
    try {
      const pageViewData = {
        eventType: 'content_page_view',
        entryId: sdk.entry.getSys().id,
        contentType: sdk.contentType.sys.id,
        userAgent: navigator.userAgent,
        timestamp: new Date().toISOString(),
        userId: sdk.user.sys.id,
        sessionId: Date.now().toString(),
      };

      console.log('👁️ Amplitude Sidebar: Page view data prepared:', pageViewData);
      
      // Simulate API call delay
      await new Promise(resolve => setTimeout(resolve, 1000));
      
      setEventCount(prev => {
        const newCount = prev + 1;
        console.log('📊 Amplitude Sidebar: Event count updated to:', newCount);
        return newCount;
      });
      
      setLastEventTime(new Date().toLocaleTimeString());
      
      console.log('✅ Amplitude Sidebar: Page view tracking completed successfully');
      sdk.notifier.success('Page view tracked in Amplitude!');
      
    } catch (error) {
      console.error('❌ Amplitude Sidebar: Error tracking page view:', error);
      sdk.notifier.error('Failed to track page view');
    } finally {
      setIsTracking(false);
      console.log('🏁 Amplitude Sidebar: Page view tracking process finished');
    }
  };

  const handleTrackCustomEvent = async () => {
    console.log('🎯 Amplitude Sidebar: Custom event tracking initiated...');
    setIsTracking(true);
    
    try {
      const customEventData = {
        eventType: 'content_interaction',
        action: 'sidebar_button_click',
        entryId: sdk.entry.getSys().id,
        contentType: sdk.contentType.sys.id,
        timestamp: new Date().toISOString(),
        userId: sdk.user.sys.id,
        properties: {
          buttonType: 'custom_tracking',
          location: 'entry_sidebar',
        }
      };

      console.log('🎯 Amplitude Sidebar: Custom event data prepared:', customEventData);
      
      // Simulate API call delay
      await new Promise(resolve => setTimeout(resolve, 800));
      
      setEventCount(prev => {
        const newCount = prev + 1;
        console.log('📊 Amplitude Sidebar: Event count updated to:', newCount);
        return newCount;
      });
      
      setLastEventTime(new Date().toLocaleTimeString());
      
      console.log('✅ Amplitude Sidebar: Custom event tracking completed successfully');
      sdk.notifier.success('Custom event tracked in Amplitude!');
      
    } catch (error) {
      console.error('❌ Amplitude Sidebar: Error tracking custom event:', error);
      sdk.notifier.error('Failed to track custom event');
    } finally {
      setIsTracking(false);
      console.log('🏁 Amplitude Sidebar: Custom event tracking process finished');
    }
  };

  console.log('🎨 Amplitude Sidebar: Rendering sidebar with event count:', eventCount);

  return (
    <Stack spacing="spacingM" flexDirection="column">
      <Card padding="spacingM">
        <Stack spacing="spacingS" flexDirection="column">
          <Paragraph>
            <strong>Amplitude Analytics</strong>
          </Paragraph>
          
          <Stack spacing="spacingXs" flexDirection="row">
            <Badge variant="secondary">Events: {eventCount}</Badge>
            {lastEventTime && (
              <Badge variant="positive">Last: {lastEventTime}</Badge>
            )}
          </Stack>
        </Stack>
      </Card>

      <Button
        variant="primary"
        isFullWidth={true}
        isLoading={isTracking}
        isDisabled={isTracking}
        onClick={handleTrackPageView}>
        {isTracking ? 'Tracking...' : 'Track Page View'}
      </Button>

      <Button
        variant="secondary"
        isFullWidth={true}
        isLoading={isTracking}
        isDisabled={isTracking}
        onClick={handleTrackCustomEvent}>
        {isTracking ? 'Tracking...' : 'Track Custom Event'}
      </Button>

      <Paragraph style={{ fontSize: '12px', color: '#6b7280' }}>
        Entry ID: {sdk.entry.getSys().id.substring(0, 8)}...
      </Paragraph>
    </Stack>
  );
};

export default Sidebar;
