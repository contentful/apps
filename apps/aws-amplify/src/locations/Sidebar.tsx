import { useEffect, useState } from 'react';
import { Button, Paragraph } from '@contentful/f36-components';
import { SidebarAppSDK } from '@contentful/app-sdk';
import { useCMA, useSDK } from '@contentful/react-apps-toolkit';
import { getItemLocalStorage, build_status_key, setItemLocalStorage } from '../lib/localStorage';

const Sidebar = () => {
  const sdk = useSDK<SidebarAppSDK>();
  const cma = useCMA();
  const [isLoading, setIsLoading] = useState(false);
  const [lastBuildInitiated, setLastBuildInitiated] = useState(
    getItemLocalStorage(build_status_key) || null
  );

  useEffect(() => {
    if (lastBuildInitiated) {
      setLastBuildInitiated(lastBuildInitiated);
    }
  }, [lastBuildInitiated]);

  const handleBuildAppActionCall = async () => {
    const buildInitiated = new Date().toLocaleTimeString();
    setIsLoading(true);
    try {
      const res = await cma.appActionCall.createWithResponse(
        {
          appActionId: 'amplifyBuildAction',
          environmentId: sdk.ids.environment,
          spaceId: sdk.ids.space,
          appDefinitionId: sdk.ids.app!,
        },
        {
          parameters: {
            amplifyWebhookUrl: sdk.parameters.installation.amplifyWebhookUrl,
          },
        }
      );
      setLastBuildInitiated(buildInitiated);
      setItemLocalStorage(build_status_key, buildInitiated);
      setIsLoading(false);
      const { message } = JSON.parse(res.response.body);
      sdk.notifier.success(message);
    } catch (err) {
      console.log({ err });
    }
  };

  return (
    <Paragraph>
      <Button
        variant="primary"
        isFullWidth={true}
        isLoading={isLoading}
        isDisabled={isLoading}
        onClick={handleBuildAppActionCall}>
        {isLoading ? 'Initiating build...' : 'Build now'}
      </Button>
      <span>Last build started: {lastBuildInitiated}</span>
    </Paragraph>
  );
};

export default Sidebar;
