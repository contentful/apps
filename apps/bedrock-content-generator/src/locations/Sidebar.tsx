import DisplaySidebarWarning from "@components/app/sidebar/DisplaySidebarWarning";
import SidebarButtons from "@components/app/sidebar/SidebarButtons";
import { useAutoResizer } from "@contentful/react-apps-toolkit";
import useSidebarParameters from "@hooks/sidebar/useSidebarParameters";

const Sidebar = () => {
  const { hasBrandProfile } = useSidebarParameters();
  useAutoResizer();

  return (
    <>
      <SidebarButtons />
      <DisplaySidebarWarning
        hasBrandProfile={hasBrandProfile}
      />
    </>
  );
};

export default Sidebar;
