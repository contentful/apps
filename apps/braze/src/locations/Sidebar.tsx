import { SidebarAppSDK } from '@contentful/app-sdk';
import { Button } from '@contentful/f36-components';
import { useAutoResizer, useSDK } from '@contentful/react-apps-toolkit';
import { EntryInfo } from './Dialog';
import { DIALOG_TITLE, SIDEBAR_BUTTON_TEXT } from '../utils';

const Sidebar = () => {
  const sdk = useSDK<SidebarAppSDK>();
  useAutoResizer();

  const invocationParams: EntryInfo = {
    id: sdk.ids.entry,
    contentTypeId: sdk.ids.contentType,
    title: sdk.entry.fields[sdk.contentType.displayField].getValue(),
  };

  return (
    <Button
      variant="primary"
      isFullWidth={true}
      onClick={() => {
        sdk.dialogs.openCurrentApp({
          title: DIALOG_TITLE,
          parameters: invocationParams,
          width: 'fullWidth',
        });
      }}>
      {SIDEBAR_BUTTON_TEXT}
    </Button>
  );
};

export default Sidebar;
