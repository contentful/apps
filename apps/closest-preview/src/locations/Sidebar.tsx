import {
  Box,
  List,
  Paragraph,
  RelativeDateTime,
  Skeleton,
  Text,
  TextLink,
} from '@contentful/f36-components';
import { ArrowSquareOutIcon } from '@contentful/f36-icons';
import { SidebarAppSDK } from '@contentful/app-sdk';
import { useAutoResizer, useSDK } from '@contentful/react-apps-toolkit';
import { EntryProps, KeyValueMap } from 'contentful-management';
import { useCallback, useEffect, useState } from 'react';

const MAX_DEPTH = 10;

const Sidebar = () => {
  const sdk = useSDK<SidebarAppSDK>();
  useAutoResizer();
  const [entries, setEntries] = useState<EntryProps[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [maxDepthReached, setMaxDepthReached] = useState<boolean>(false);
  const defaultLocale = sdk.locales.default;

  const getRelatedEntries = useCallback(
    async (id: string): Promise<EntryProps[]> => {
      try {
        const response = await sdk.cma.entry.getMany({
          query: {
            links_to_entry: id,
            order: '-sys.updatedAt',
            limit: 5,
          },
          spaceId: sdk.ids.space,
          environmentId: sdk.ids.environment,
        });

        return response.items;
      } catch (error) {
        console.error(error);
        return [];
      }
    },
    [sdk.ids.space, sdk.ids.environment]
  );

  const splitEntriesFromRoot = (
    entry: EntryProps<KeyValueMap>,
    checkedEntries: Set<string>,
    rootEntryData: EntryProps[]
  ) => {
    const entryId = entry?.sys?.id;
    if (!entryId || checkedEntries.has(entryId)) {
      return false;
    }

    checkedEntries.add(entryId);
    const slug = entry.fields.slug?.[defaultLocale];
    if (slug) {
      rootEntryData.push(entry);
      return false;
    }

    return true;
  };

  const getRootEntries = useCallback(
    async (id: string): Promise<{ entries: EntryProps[]; maxDepthReached: boolean }> => {
      const rootEntryData: EntryProps[] = [];
      let childEntries: EntryProps[] = [];
      const checkedEntries: Set<string> = new Set([id]);
      let depth = 0;
      const maxDepth = MAX_DEPTH;
      let depthReached = false;

      try {
        const initialEntry = await sdk.cma.entry.get({
          entryId: id,
          spaceId: sdk.ids.space,
          environmentId: sdk.ids.environment,
        });
        childEntries = [initialEntry];
      } catch (error) {
        console.error('Failed to fetch initial entry:', error);
        return { entries: [], maxDepthReached: false };
      }

      while (childEntries.length > 0 && depth < maxDepth) {
        const relatedEntries = await Promise.all(
          childEntries.map((entry) => getRelatedEntries(entry.sys.id))
        );

        childEntries = relatedEntries.flatMap((rEntry) =>
          rEntry.filter((item: EntryProps) => {
            return splitEntriesFromRoot(item, checkedEntries, rootEntryData);
          })
        );

        depth++;
      }

      if (depth >= maxDepth && rootEntryData.length === 0) {
        depthReached = true;
      }

      return { entries: rootEntryData, maxDepthReached: depthReached };
    },
    [getRelatedEntries, sdk.ids.space, sdk.ids.environment, defaultLocale]
  );

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      const result = await getRootEntries(sdk.ids.entry);

      setEntries(result.entries);
      setMaxDepthReached(result.maxDepthReached);
      setIsLoading(false);
    };
    fetchData();
  }, [getRootEntries, sdk.ids.entry]);

  if (isLoading) {
    return (
      <Skeleton.Container>
        <Skeleton.BodyText numberOfLines={3} />
      </Skeleton.Container>
    );
  }

  if (maxDepthReached) {
    return <Text>Max depth of {MAX_DEPTH} reached for entry.</Text>;
  }

  return (
    <List>
      {entries.map((entry: EntryProps) => {
        const entryLink = `https://${sdk.hostnames.webapp}/spaces/${sdk.ids.space}/environments/${sdk.ids.environment}/entries/${entry.sys.id}`;
        return (
          <Box key={entry.sys.id} marginBottom="spacingS">
            <List.Item key={entry.sys.id}>
              <TextLink
                href={entryLink}
                target="_blank"
                rel="noopener noreferrer"
                icon={<ArrowSquareOutIcon />}
                alignIcon="end">
                {entry.fields.title?.[defaultLocale] || entry.sys.id.slice(0, 8)}
              </TextLink>
              <br />
              <Paragraph fontSize="fontSizeM" fontColor="gray500" fontWeight="fontWeightMedium">
                Updated <RelativeDateTime date={entry.sys.updatedAt} />
              </Paragraph>
            </List.Item>
          </Box>
        );
      })}
    </List>
  );
};

export default Sidebar;
