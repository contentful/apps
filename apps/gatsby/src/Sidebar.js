import React from 'react';
import PropTypes from 'prop-types';
import { ExtensionUI } from '@gatsby-cloud-pkg/gatsby-cms-extension-base';
import {
  Spinner,
  Paragraph,
  HelpText,
  Icon,
  ValidationMessage
} from '@contentful/forma-36-react-components';

const STATUS_STYLE = { textAlign: 'center', color: '#7f7c82' };
const ICON_STYLE = { marginBottom: '-4px' };

const callWebhook = (webhookUrl, authToken) => fetch(webhookUrl, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'x-preview-update-source': 'contentful-sidebar-extension',
    'x-preview-auth-token': authToken || ''
  },
  body: JSON.stringify({})
});

export default class Sidebar extends React.Component {
  static propTypes = {
    sdk: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);

    this.state = {};
    this.sdk = props.sdk;
    this.sdk.entry.onSysChanged(this.onSysChanged);
  }

  onSysChanged = () => {
    if (this.debounceInterval) {
      clearInterval(this.debounceInterval);
    }
    this.debounceInterval = setInterval(this.refreshPreview, 1000);
  };

  componentDidMount() {
    this.sdk.window.startAutoResizer();
  }

  refreshPreview = async () => {
    if (this.debounceInterval) {
      clearInterval(this.debounceInterval);
    }

    const { webhookUrl, authToken } = this.sdk.parameters.installation;

    if (!webhookUrl) {
      return;
    }

    this.setState({ busy: true })

    const [res] = await Promise.all([
      // Convert any errors thrown to non-2xx HTTP response
      // (for uniform handling of errors).
      callWebhook(webhookUrl, authToken).catch(() => ({ ok: false })),
      // Make sure the spinner spins for at least a second
      // (to avoid a blink of text).
      new Promise(resolve => setTimeout(resolve, 1000))
    ]);

    this.setState({ busy: false, ok: res.ok });
  };

  render = () => {
    const { webhookUrl, previewUrl, authToken } = this.sdk.parameters.installation;
    const contentSlug = this.sdk.entry.fields.slug;

    return (
      <div className="extension">
        <div className="flexcontainer">
          <ExtensionUI
            contentSlug={contentSlug && contentSlug.getValue()}
            previewUrl={previewUrl}
            authToken={authToken}
          />
          {webhookUrl && this.renderRefreshStatus()}
        </div>
      </div>
    );
  };

  renderRefreshStatus = () => {
    const { busy, ok } = this.state;

    return (
      <HelpText style={STATUS_STYLE}>
        {busy && (
          <>
            <Spinner />
            {' '}Sending entry data...
          </>
        )}
        {!busy && (ok === true) && (
          <>
            <Icon icon="CheckCircle" color="positive" style={ICON_STYLE} />
            {' '}Entry data in Gatsby up to date!
          </>
        )}
        {!busy && (ok === false) && (
          <>
            <Icon icon="Warning" color="negative" style={ICON_STYLE} />
            {' '}Last update failed.
          </>
        )}
      </HelpText>
    );
  };
}
