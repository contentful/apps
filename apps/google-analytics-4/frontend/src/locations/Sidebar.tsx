import AnalyticsApp from 'components/main-app/AnalyticsApp/AnalyticsApp';
import { useSDK } from '@contentful/react-apps-toolkit';
import { SidebarExtensionSDK } from '@contentful/app-sdk';
import { useApi } from 'hooks/useApi';
import { AppInstallationParameters } from 'types';

const Sidebar = () => {
  const sdk = useSDK<SidebarExtensionSDK>();

  const { serviceAccountKeyId, propertyId, contentTypes } = sdk.parameters
    .installation as AppInstallationParameters;

  const currentContentType = sdk.contentType.sys.id;
  const slugFieldInfo = contentTypes[currentContentType] ?? { slugField: '', urlPrefix: '' };

  const api = useApi(serviceAccountKeyId);

  return (
    <>
      <AnalyticsApp api={api} propertyId={propertyId} slugFieldInfo={slugFieldInfo} />
    </>
  );
};

export default Sidebar;
