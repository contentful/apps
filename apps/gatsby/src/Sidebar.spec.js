/* global global */
import React from 'react';
import { render, cleanup, wait } from '@testing-library/react';

import Sidebar from './Sidebar';

const mockSdk = {
  location: {
    is: val => val === 'entry-sidebar'
  },
  parameters: {
    installation: {
      previewUrl: 'https://preview.com',
      webhookUrl: 'https://webhook.com',
      authToken: 'test-token'
    }
  },
  entry: {
    onSysChanged: jest.fn(),
    fields: {
      slug: {
        getValue: jest.fn(() => 'preview-slug')
      }
    }
  },
  window: {
    startAutoResizer: jest.fn()
  }
};

describe('Gatsby App Sidebar', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = render(<Sidebar sdk={mockSdk} />);

    expect(container).toMatchSnapshot();
  });

  it('should debounce the fetch', async () => {
    const mockFetch = jest.fn(() => Promise.resolve());
    let timerComplete = false;
    global.fetch = mockFetch;
    mockSdk.entry.onSysChanged.mockImplementationOnce(fn => {
      fn();
      setTimeout(() => {
        timerComplete = true;
      }, 1000);
      return jest.fn();
    });

    render(<Sidebar sdk={mockSdk} />);

    await wait(
      () => {
        if (timerComplete) {
          return;
        }
        throw 0;
      },
      { timeout: 5000 }
    );

    expect(mockFetch).toHaveBeenCalledWith('https://webhook.com', {
      body: '{}',
      headers: {
        'Content-Type': 'application/json',
        'x-preview-auth-token': 'test-token',
        'x-preview-update-source': 'contentful-sidebar-extension'
      },
      method: 'POST'
    });
  });
});
