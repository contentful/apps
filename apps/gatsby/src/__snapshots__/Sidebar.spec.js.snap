// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gatsby App Sidebar should match snapshot 1`] = `
<div>
  <div
    class="extension"
  >
    <div
      class="flexcontainer"
    >
      <div>
        <button
          class="preview-button css-w7g5y4"
          type="button"
        >
          Open Preview
        </button>
        <div
          class="powered-by css-1r59ijf"
        >
          <p
            class="css-1kyb6cu"
          >
            Powered by:
          </p>
          <svg
            fill="none"
            height="40"
            viewBox="0 0 24 24"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.4286 12H15.4286V13.7143H19.5429C18.9429 16.2857 17.0572 18.4286 14.5715 19.2857L4.71433 9.42855C5.7429 6.42855 8.65719 4.2857 12 4.2857C14.5715 4.2857 16.8858 5.57141 18.3429 7.54284L19.6286 6.42855C17.9143 4.11427 15.1715 2.57141 12 2.57141C7.5429 2.57141 3.77147 5.74284 2.82861 9.94284L14.1429 21.2571C18.2572 20.2286 21.4286 16.4571 21.4286 12Z"
              fill="white"
            />
            <path
              d="M2.57153 12.0857C2.57153 14.4857 3.51439 16.8 5.31439 18.6C7.11439 20.4 9.51439 21.3428 11.8287 21.3428L2.57153 12.0857Z"
              fill="white"
            />
            <path
              clip-rule="evenodd"
              d="M12 0C5.4 0 0 5.4 0 12C0 18.6 5.4 24 12 24C18.6 24 24 18.6 24 12C24 5.4 18.6 0 12 0ZM5.31429 18.6857C3.51429 16.8857 2.57143 14.4857 2.57143 12.1714L11.9143 21.4286C9.51429 21.3429 7.11429 20.4857 5.31429 18.6857ZM2.82857 9.94286L14.0571 21.1714C18.2571 20.2286 21.4286 16.4571 21.4286 12H15.4286V13.7143H19.5429C18.9429 16.2857 17.0571 18.4286 14.5714 19.2857L4.71429 9.42857C5.82857 6.42857 8.65714 4.28571 12 4.28571C14.5714 4.28571 16.8857 5.57143 18.3429 7.54286L19.6286 6.42857C17.9143 4.11429 15.1714 2.57143 12 2.57143C7.54286 2.57143 3.77143 5.74286 2.82857 9.94286Z"
              fill="#663399"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <p
        class="HelpText__HelpText___uWbja"
        data-test-id="cf-ui-help-text"
        style="text-align: center; color: rgb(127, 124, 130);"
      />
    </div>
  </div>
</div>
`;
